### ISSUE TITLE
Question about performance when creating operations

### ISSUE THREAD

**Issue by sfontana** [googleads/googleads-python-lib#118]
Hello,

What is the benefit of creating XML from a service object in https://github.com/googleads/googleads-python-lib/blob/master/googleads/adwords.py#L740
````
      service = self._client.GetService(operation.service, self._version)
      service.suds_client.set_options(nosend=True)
      service_request = (getattr(service, operation.method)
                         (operations).envelope)
      service.suds_client.set_options(nosend=False)
      return service_request
````

only to parse it again in https://github.com/googleads/googleads-python-lib/blob/master/googleads/adwords.py#L772

````
root = ElementTree.fromstring(raw_request_xml)
      return root.find('{http://schemas.xmlsoap.org/soap/envelope/}Body').find(
          './/')
````

Is there a better way of doing this?

**Comment by msaniscalchi** [comment]
Hello,

That's a good question!

**tl;dr:** I wouldn't rule out that there _could_ be a better way to do certain tasks. If you happen to have any feedback to improve the implementation, I'd be happy to hear you out.

To better explain the current implementation, let's say you're using BatchJobHelper and passed a big list of operations to the UploadOperations method--e.g. the big set of operations found in the [add_complete_campaigns_using_batch_job.py example](https://github.com/googleads/googleads-python-lib/blob/master/examples/adwords/v201603/campaign_management/add_complete_campaigns_using_batch_job.py).

We need to generate the request body XML that actually performs all of the operations in a single request when uploaded to the given upload_url. But wait! The operations sent in span across multiple services, so producing the request body with suds alone isn't actually possible. As a result, for each list of operations we:

1. Generate the raw XML from the corresponding service.
1. Extract the method element and its contents from the raw XML.
1. Perform additional tweaks for compatibility with BatchJobService.

Once all of the XML for the operations are generated, they are joined together and then added to a template to produce the request body.

I'm going to close this issue because I think I've sufficiently answered your question, but feel free to reopen if you have any feedback you'd like to share.
